.github/workflows/ci.yml

```diff
@@ -0,0 +1,26 @@
+name: ci
+
+on:
+    pull_request:
+        branches:
+            - main
+    push:
+        branches:
+            - main
+
+jobs:
+    unit-tests:
+        name: Run Tests
+        runs-on: ubuntu-latest
+        steps:
+            - name: Check out code
+              uses: actions/checkout@v3
+
+            - name: Setup node
+              uses: actions/setup-node@v3
+              with:
+                  node-version: '20'
+                  cache: 'npm'
+
+            - name: Install node dependencies and run Tests
+              run: npm install-ci-test
```

Comment:
Could we split this into two steps, `npm ci` and `npm test`? I realise this does the same, but having it split gives us more room to do other things if needed